plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/RefactorElementPropertyTester.java

```diff
@@ -31,9 +35,25 @@ protected boolean isSupported(final Object element) {
 			return isSupported(editPart.getModel());
 		}
 
-		return ((element instanceof final IInterfaceElement ie && !(element instanceof ErrorMarkerInterface))
-				&& (ie.getFBNetworkElement() instanceof final FB fb
-						&& !fb.getType().getTypeEntry().getFile().isReadOnly()))
-				|| (element instanceof final FB fbb && fbb.eContainer() instanceof BaseFBType);
+		if (element instanceof final EObject eObject
+				&& EcoreUtil.getRootContainer(eObject) instanceof final LibraryElement libraryElement
+				&& libraryElement.getTypeEntry().getFile().isReadOnly()) {
+			return false;
+		}
+
+		if (element instanceof final IInterfaceElement ie && !(ie instanceof ErrorMarkerInterface)) {
+			if (ie.getFBNetworkElement() != null && ie.getFBNetworkElement().getType() != null
+					&& ie.getFBNetworkElement().getType().getTypeEntry() != null
+					&& ie.getFBNetworkElement().getType().getTypeEntry().getFile() != null) {
+				return !ie.getFBNetworkElement().getType().getTypeEntry().getFile().isReadOnly();
```

Comment:
As mentioned before, you can directly get the type entry from the FB network element:
```suggestion
			if (ie.getFBNetworkElement() != null && ie.getFBNetworkElement().getTypeEntry() != null
					&& ie.getFBNetworkElement().getTypeEntry().getFile() != null) {
				return !ie.getFBNetworkElement().getTypeEntry().getFile().isReadOnly();
```
This is not only shorter but also avoids potentially (re-)loading the type unnecessarily.